bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/link/ItemChannelLinkConfigDescriptionProvider.java

```diff
@@ -90,21 +86,41 @@ public ConfigDescription getConfigDescription(URI uri, Locale locale) {
     }
 
     private List<ParameterOption> getOptions(ItemChannelLink link, Item item, Channel channel, Locale locale) {
-        return Stream
-                .concat(Stream.of(defaultProfileFactory.getApplicableProfileTypeUIDs(link, item, channel)),
-                        profileAdvisors.stream().map(f -> f.getApplicableProfileTypeUIDs(link, item, channel)))
-                .flatMap(c -> c.stream())
-                .map(profileTypeUID -> new ParameterOption(profileTypeUID.toString(), profileTypeUID.getLabel()))
+        Collection<ProfileType> profileTypes = profileTypeRegistry.getAll();
+        return profileTypes.stream().filter(profileType -> {
+            switch (channel.getKind()) {
+                case STATE:
+                    return profileType instanceof StateProfileType && isSupportedItemType(profileType, item);
+                case TRIGGER:
+                    return profileType instanceof TriggerProfileType && isSupportedItemType(profileType, item)
+                            && isSupportedChannelType(profileType, channel);
+                default:
+                    throw new IllegalArgumentException("Unknown channel kind: " + channel.getKind());
+            }
+        }).map(profileType -> new ParameterOption(profileType.getUID().toString(), profileType.getLabel()))
                 .collect(Collectors.toList());
     }
 
-    @Reference(cardinality = ReferenceCardinality.MULTIPLE, policy = ReferencePolicy.DYNAMIC)
-    public void addProfileAdvisor(ProfileAdvisor profileAdvisor) {
-        profileAdvisors.add(profileAdvisor);
+    private boolean isSupportedItemType(ProfileType profileType, Item item) {
+        return profileType.getSupportedItemTypes() == ProfileType.ANY_ITEM_TYPE
+                || profileType.getSupportedItemTypes().contains(item.getType());
+    }
+
+    private boolean isSupportedChannelType(ProfileType profileType, Channel channel) {
+        return profileType instanceof StateProfileType
```

Comment:
`profileType instanceof TriggerProfileType`?

Reply:
No I think that StateProfileType is what is meant here, but I don't quite understand why a StateProfileType would always support all ChannelTypes. Could there not be profiles that only support a specific channel type?